builder.json

```diff
@@ -20,19 +20,60 @@
       }
     }
   },
-  "test_steps": [
+  "build_steps": [
+    "build",
     [
       "mkdir",
       "build_shadow"
     ],
     [
-      "cmake", 
-      "--version"
+      "mkdir",
+      "build_basic_pubsub"
+    ],
```

Comment:
trivial: make each directory before its related build step, instead of a bunch of makes then bunch of builds